core/src/main/java/org/sonarsource/sonarlint/core/telemetry/TelemetryAnalyzerPerformance.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * SonarLint Core - Implementation
+ * Copyright (C) 2009-2018 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonarsource.sonarlint.core.telemetry;
+
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.TreeMap;
+
+public class TelemetryAnalyzerPerformance {
+  private static final TreeMap<Integer, String> INTERVALS;
+
+  static {
+    INTERVALS = new TreeMap<>();
+    INTERVALS.put(300, "0-300");
+    INTERVALS.put(500, "300-500");
+    INTERVALS.put(1000, "500-1000");
+    INTERVALS.put(2000, "1000-2000");
+    INTERVALS.put(4000, "2000-4000");
+    INTERVALS.put(Integer.MAX_VALUE, "4000+");
+  }
+
+  private Map<String, Integer> frequencies;
+
+  public TelemetryAnalyzerPerformance() {
+    frequencies = new LinkedHashMap<>();
+    INTERVALS.forEach((k, v) -> frequencies.put(v, 0));
+  }
+  
+  public void registerAnalysis(int analysisTimeMs) {
+    Entry<Integer, String> entry = INTERVALS.higherEntry(analysisTimeMs);
+    if (entry != null) {
+      frequencies.compute(entry.getValue(), (k, v) -> v != null ? v + 1 : 1);
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Add parentheses to make the operator precedence explicit. ([squid:S864](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS864))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.sonarlint.core%3Asonarlint-core-parent&branch=telemetry_perf&issues=AWHYGJP7wd3JK8wdtRez&open=AWHYGJP7wd3JK8wdtRez)

[ISSUE_KEY]: AWHYGJP7wd3JK8wdtRez